### PR TITLE
feat: Preapare the connection signature raw data as per the spec

### PR DESCRIPTION
**Pull request by talwinder50**
closes: #345

Signed-off-by: talwinder.kaur <talwinder.kaur@securekey.com>

### CODE REVIEW

**Comment by sandrask** [comment]
Please update your PR with exact description. There is more functionality mentioned in your PR. Also, if you are implementing something per spec, please mentioned which spec and where in the spec.

**Comment by talwinder50** [comment]
> Please update your PR with exact description. There is more functionality mentioned in your PR. Also, if you are implementing something per spec, please mentioned which spec and where in the spec.

File: pkg/didcomm/protocol/didexchange/states_test.go [hunk 1]
```diff
@@ -516,7 +516,6 @@ func TestPrepareConnectionSignature(t *testing.T) {
 		connectionSignature, err := prepareConnectionSignature(connection)
 		require.NoError(t, err)
 		require.NotNil(t, connectionSignature)
-		require.Equal(t, connectionSignature.Type, "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/signature/1.0/ed25519Sha512_single")
```
Comment (sandrask) [inline pkg/didcomm/protocol/didexchange/states_test.go]: Add test to make sure that signature data equals original DID after unmarshal.

**Comment by talwinder50** [comment]
closed by mistake

**Pull request closed without merging**